### PR TITLE
Custom Pages for `<UserProfile />` and `<OrganizationProfile />`

### PR DESCRIPTION
**Pull request by anagstef**
This PR adds documentation for the [Custom pages on UserProfile/OrganizationProfile](https://github.com/clerkinc/javascript/pull/1822).

🔎 [Preview](https://docs-preview-404.clerkpreview.com/docs)

Pages with changes:
https://docs-preview-404.clerkpreview.com/docs/references/javascript/clerk/user-profile
https://docs-preview-404.clerkpreview.com/docs/references/javascript/clerk/organization-profile

New pages:
https://docs-preview-404.clerkpreview.com/docs/references/javascript/types/custom-page
https://docs-preview-404.clerkpreview.com/docs/components/customization/user-profile
https://docs-preview-404.clerkpreview.com/docs/components/customization/organization-profile

### CODE REVIEW

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 1]
```diff
@@ -0,0 +1,418 @@
+---
+title: Customize Organization Profile
+description: How to add custom pages to the `<OrganizationProfile />` component
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:3]:
```suggestion
description: Learn how to add custom pages and include external links within the navigation sidebar of the <OrganizationProfile /> component.
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 2]
```diff
@@ -0,0 +1,418 @@
+---
+title: Customize Organization Profile
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:2]:
```suggestion
title: <OrganizationProfile /> customization
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+description: How to add custom pages to the `<OrganizationProfile />` component
+---
+
+# Custom pages in Organization Profile
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:6]:
```suggestion
# `<OrganizationProfile />`

The `<OrganizationProfile />` component supports the addition of custom pages and use of external links in the navigation sidebar. 
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+
+## `<OrganizationProfile.Page />`
+
+The `<OrganizationProfile />` component supports the addition of custom pages. These custom pages can be rendered inside the `<OrganizationProfile />` component and provide a way to incorporate app-specific settings or additional functionality.
+
+To add a custom page, use the `<OrganizationProfile.Page />` component.
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:12]:
```suggestion
Custom pages can be rendered inside the `<OrganizationProfile />` component and provide a way to incorporate app-specific settings or additional functionality.

To add a custom page to the `<OrganizationProfile />` component, use the `<OrganizationProfile.Page />` component.
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+
+### Usage
+
+<Tabs items={['Next.js', 'React', 'Remix']}>
+    <Tab>
+        `<OrganizationProfile.Page />` can be rendered **only on the client**, and in Next.js app-router projects, they need to be under the `"use client";` flag.
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:18]:
```suggestion
        `<OrganizationProfile.Page />` can be rendered **only on the client**. For Next.js applications using App Router, the `"use client";` flag must be added.
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+
+### Props
+
+All props below are required.
+
+| Name | Type | Description |
+| --- | --- | --- |
+| `label` | `string` | The name that will be displayed in the navigation sidebar for the custom page. |
+| `labelIcon` | `React.ReactElement` | An icon displayed next to the label in the navigation sidebar. |
+| `url` | `string` | The path segment that will be used to navigate to the custom page. (e.g. if the `OrganizationProfile` component is rendered at `/organization`, then the custom page will be accessed at `/organization/{url}` when using path routing) |
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:114]:
```suggestion
| `url` | `string` | The path segment that will be used to navigate to the custom page. (e.g. if the `OrganizationProfile` component is rendered at `/organization`, then the custom page will be accessed at `/organization/{url}` when using path routing). |
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+| `children` | `React.ReactElement` | The components to be rendered as content inside the custom page. |
+
+
+
+
+## `<OrganizationProfile.Link />`
+
+In addition to custom pages, you can add external links to the `<OrganizationProfile />` navigation sidebar using the `<OrganizationProfile.Link />` component.
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:122]:
```suggestion
You can add external links to the `<OrganizationProfile />` navigation sidebar using the `<OrganizationProfile.Link />` component.
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+
+### Usage
+
+<Tabs items={['Next.js', 'React', 'Remix']}>
+    <Tab>
+        `<OrganizationProfile.Link />` can be rendered **only on the client**, and in Next.js app-router projects, they need to be under the `"use client";` flag.
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:128]:
```suggestion
        `<OrganizationProfile.Link />` can be rendered **only on the client**. For Next.js applications using App Router, the `"use client";` flag must be added.
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+
+### Props
+
+All props below are required.
+
+| Name | Type | Description |
+| --- | --- | --- |
+| `label` | `string` | The name that will be displayed in the navigation sidebar for the link.  |
+| `labelIcon` | `React.ReactElement` | An icon displayed next to the label in the navigation sidebar. |
+| `url` | `string` | The absolute or relative url to navigate to |
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:201]:
```suggestion
| `url` | `string` | The absolute or relative url to navigate to. |
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 10]
```diff
@@ -0,0 +1,418 @@
...
+
+
+## Advanced use cases
+
+### Reordering Default Routes
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:206]:
```suggestion
### Reordering default routes
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 11]
```diff
@@ -0,0 +1,418 @@
...
+
+If you want to reorder the default routes (`Members` and `Settings`) in the `OrganizationProfile` navigation sidebar, you can use the `<OrganizationProfile.Page />` component with the `label` prop set to `'members'` or `'settings'`. This will target the existing default page and allow you to rearrange it.
+
+### Usage
+
+<Tabs items={['Next.js', 'React', 'Remix']}>
+    <Tab>
+        `<OrganizationProfile.Page />` and `<OrganizationProfile.Link />` can be rendered **only on the client**, and in Next.js app-router projects, they need to be under the `"use client";` flag.
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:214]:
```suggestion
        `<OrganizationProfile.Page />` and `<OrganizationProfile.Link />` can be rendered **only on the client**.  For Next.js applications that use App Router, the `"use client";` flag needs to be used.
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 12]
```diff
@@ -0,0 +1,418 @@
...
+
+1. Custom Page
+2. Homepage
+3. Members
+4. Settings
+
+**Notes**
+
+- The first page in the list will be rendered under the root path `/` (its `url` will be ignored) and the Clerk pages will be rendered under the path `/organitzation-members` and `/organization-settings`.
+- The first item in the list cannot be a `<OrganizationProfile.Link />` component.
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:312]:
```suggestion
It's important to note that the first page in the list will be rendered under the root path `/` (its `url` will be ignored) and the Clerk pages will be rendered under the path `/organitzation-members` and `/organization-settings`. Also, the first item in the list cannot be a `<OrganizationProfile.Link />` component.
```

File: docs/components/customization/organization-profile-custom-pages.mdx [hunk 13]
```diff
@@ -0,0 +1,418 @@
...
+
+### Using custom pages with the `<OrganizationSwitcher />` component
+
+If you are using the `<OrganizationSwitcher />` component with the default props (where the `OrganizationProfile` opens as a modal), then you should also be providing these custom pages as children to the component (using the `<OrganizationSwitcher.OrganizationProfilePage />` and `<OrganizationSwitcher.OrganizationProfileLink />` components respectively).
+
+### Usage
+
+<Tabs items={['Next.js', 'React', 'Remix']}>
+    <Tab>
+        `<OrganizationSwitcher.OrganizationProfilePage />` and `<OrganizationSwitcher.OrganizationProfileLink />` can be rendered **only on the client**, and in Next.js app-router projects, they need to be under the `"use client";` flag.
```
Comment (alexisintech) [inline docs/components/customization/organization-profile-custom-pages.mdx:322]:
```suggestion
        `<OrganizationSwitcher.OrganizationProfilePage />` and `<OrganizationSwitcher.OrganizationProfileLink />` can be rendered **only on the client**.  For Next.js applications using App Router, the `"use client";` flag must be added.
```

File: docs/components/customization/user-profile-custom-pages.mdx [hunk 14]
```diff
@@ -0,0 +1,418 @@
+---
+title: Customize User Profile
```
Comment (alexisintech) [inline docs/components/customization/user-profile-custom-pages.mdx:2]:
```suggestion
title: <UserProfile /> customization
```

File: docs/components/customization/user-profile-custom-pages.mdx [hunk 15]
```diff
@@ -0,0 +1,418 @@
...
+description: How to add custom pages to the `<UserProfile />` component
```
Comment (alexisintech) [inline docs/components/customization/user-profile-custom-pages.mdx:3]:
```suggestion
description: Learn how to add custom pages and include external links within the navigation sidebar of the <UserProfile /> component.
```

File: docs/components/customization/user-profile-custom-pages.mdx [hunk 16]
```diff
@@ -0,0 +1,418 @@
...
+---
+
+# Custom pages in User Profile
+
+## `<UserProfile.Page />`
+
+The `<UserProfile />` component supports the addition of custom pages. These custom pages can be rendered inside the `<UserProfile />` component and provide a way to incorporate app-specific settings or additional functionality.
+
+To add a custom page, use the `<UserProfile.Page />` component.
```
Comment (alexisintech) [inline docs/components/customization/user-profile-custom-pages.mdx:12]:
```suggestion
# `<UserProfile />` customization

The `<UserProfile />` component supports the addition of custom pages and use of external links in the navigation sidebar. 

## `<UserProfile.Page />`

Custom pages can be rendered inside the `<UserProfile />` component and provide a way to incorporate app-specific settings or additional functionality.

To add a custom page to the `<UserProfile />` component, use the `<UserProfile.Page />` component.
```

File: docs/components/customization/user-profile-custom-pages.mdx [hunk 17]
```diff
@@ -0,0 +1,418 @@
...
+
+### Usage
+
+<Tabs items={['Next.js', 'React', 'Remix']}>
+    <Tab>
+        `<UserProfile.Page />` can be rendered **only on the client**, and in Next.js app-router projects, they need to be under the `"use client";` flag.
```
Comment (alexisintech) [inline docs/components/customization/user-profile-custom-pages.mdx:18]:
```suggestion
        `<UserProfile.Page />` can be rendered **only on the client**. For Next.js applications using App Router, the `"use client";` flag must be added.
```

File: docs/components/customization/user-profile-custom-pages.mdx [hunk 18]
```diff
@@ -0,0 +1,418 @@
...
+
+## `<UserProfile.Link />`
+
+In addition to custom pages, you can add external links to the `<UserProfile />` navigation sidebar using the `<UserProfile.Link />` component.
+
+### Usage
+
+<Tabs items={['Next.js', 'React', 'Remix']}>
+    <Tab>
+        `<UserProfile.Link />` can be rendered **only on the client**, and in Next.js app-router projects, they need to be under the `"use client";` flag.
```
Comment (alexisintech) [inline docs/components/customization/user-profile-custom-pages.mdx:128]:
```suggestion
        `<UserProfile.Link />` can be rendered **only on the client**. For Next.js applications using App Router, the `"use client";` flag must be added.
```

File: docs/components/customization/user-profile-custom-pages.mdx [hunk 19]
```diff
@@ -0,0 +1,418 @@
...
+
+### Reordering Default Routes
+
+If you want to reorder the default routes (`Account` and `Security`) in the `UserProfile` navigation sidebar, you can use the `<UserProfile.Page />` component with the `label` prop set to `'account'` or `'security'`. This will target the existing default page and allow you to rearrange it.
+
+### Usage
+
+<Tabs items={['Next.js', 'React', 'Remix']}>
+    <Tab>
+        `<UserProfile.Page />` and `<UserProfile.Link />` can be rendered **only on the client**, and in Next.js app-router projects, they need to be under the `"use client";` flag.
```
Comment (alexisintech) [inline docs/components/customization/user-profile-custom-pages.mdx:214]:
```suggestion
        `<UserProfile.Page />` and `<UserProfile.Link />` can be rendered **only on the client**. For Next.js applications using App Router, the `"use client";` flag must be added.
```

File: docs/components/customization/user-profile-custom-pages.mdx [hunk 20]
```diff
@@ -0,0 +1,418 @@
...
+
+### Using custom pages with the `<UserButton />` component
+
+If you are using the `<UserButton />` component with the default props (where the `UserProfile` opens as a modal), then you should also be providing these custom pages as children to the component (using the `<UserButton.UserProfilePage />` and `<UserButton.UserProfileLink />` components respectively).
+
+### Usage
+
+<Tabs items={['Next.js', 'React', 'Remix']}>
+    <Tab>
+        `<UserButton.UserProfilePage />` and `<UserButton.UserProfileLink />` can be rendered **only on the client**, and in Next.js app-router projects, they need to be under the `"use client";` flag.
```
Comment (alexisintech) [inline docs/components/customization/user-profile-custom-pages.mdx:322]:
```suggestion
        `<UserButton.UserProfilePage />` and `<UserButton.UserProfileLink />` can be rendered **only on the client**. For Next.js applications using App Router, the `"use client";` flag must be added.
```

File: docs/components/customization/user-profile-custom-pages.mdx [hunk 21]
```diff
@@ -0,0 +1,418 @@
...
+| `labelIcon` | `React.ReactElement` | An icon displayed next to the label in the navigation sidebar. |
+| `url` | `string` | The path segment that will be used to navigate to the custom page. (e.g. if the `UserProfile` component is rendered at `/user`, then the custom page will be accessed at `/user/{url}` when using path routing) |
+| `children` | `React.ReactElement` | The components to be rendered as content inside the custom page. |
+
+
+
+
+## `<UserProfile.Link />`
+
+In addition to custom pages, you can add external links to the `<UserProfile />` navigation sidebar using the `<UserProfile.Link />` component.
```
Comment (alexisintech) [inline docs/components/customization/user-profile-custom-pages.mdx:122]:
```suggestion
You can add external links to the `<UserProfile />` navigation sidebar using the `<UserProfile.Link />` component.
```

File: docs/components/customization/user-profile-custom-pages.mdx [hunk 22]
```diff
@@ -0,0 +1,418 @@
...
+
+1. Custom Page
+2. Homepage
+3. Account
+4. Security
+
+**Notes**
+
+- The first page in the list will be rendered under the root path `/` (its `url` will be ignored) and the Clerk pages will be rendered under the path `/account`.
+- The first item in the list cannot be a `<UserProfile.Link />` component.
```
Comment (alexisintech) [inline docs/components/customization/user-profile-custom-pages.mdx:312]:
```suggestion
It's important to note that the first page in the list will be rendered under the root path `/` (its `url` will be ignored) and the Clerk pages will be rendered under the path `/account`. Also, the first item in the list cannot be a `<UserProfile.Link />` component.
```

File: docs/manifest.json [hunk 23]
```diff
@@ -216,6 +216,13 @@
         ]
       ],
       ["Localization prop (i18n)", "/components/customization/localization"],
+      [
+        { "title": "Custom Pages" },
+        [
+          ["User Profile", "/components/customization/user-profile-custom-pages"],
```
Comment (alexisintech) [inline docs/manifest.json:222]:
```suggestion
          ["<UserProfile />", "/components/customization/user-profile-custom-pages"],
```

File: docs/manifest.json [hunk 24]
```diff
@@ -216,6 +216,13 @@
...
+          ["Organization Profile", "/components/customization/organization-profile-custom-pages"]
```
Comment (alexisintech) [inline docs/manifest.json:223]:
```suggestion
          ["<OrganizationProfile />", "/components/customization/organization-profile-custom-pages"]
```

File: docs/references/javascript/clerk/organization-profile.mdx [hunk 25]
```diff
@@ -236,6 +236,102 @@ Closes the organization profile overlay.
 function closeOrganizationProfile(): void;
 ```
 
+## Custom Pages
+
+Organization profile can be customized by adding custom pages to it. Custom pages are added to the organization profile using the `customPages` prop.
```
Comment (alexisintech) [inline docs/references/javascript/clerk/organization-profile.mdx:241]:
```suggestion
You can add customized pages for the `<OrganizationProfile />` component by using the `customPages` prop.

```

**Review by alexisintech — CHANGES_REQUESTED** [review]
I would rename the `organization-profile-custom-pages` file and `user-profile-custom-pages` file to simply be `organization-profile` and `user-profile`
because they already have the `/customization` path in the file name

**Review by alexisintech — APPROVED** [review]
great teamwork!!! 😸💖

File: docs/components/customization/organization-profile.mdx [hunk 26]
```diff
@@ -0,0 +1,416 @@
...
+
+Custom pages can be rendered inside the [`<OrganizationProfile />`][orgprofile-ref] component and provide a way to incorporate app-specific settings or additional functionality.
+
+To add a custom page to the [`<OrganizationProfile />`][orgprofile-ref] component, use the `<OrganizationProfile.Page />` component.
+
+### Usage
+
+<Tabs items={['Next.js', 'React', 'Remix']}>
+  <Tab>
+    `<OrganizationProfile.Page />` can be rendered **only on the client**. For Next.js applications using App Router, the `"use client";` flag must be added.
```
Comment (LekoArts) [inline docs/components/customization/organization-profile.mdx:20]:
```suggestion
    `<OrganizationProfile.Page />` can be rendered **only on the client**. For Next.js applications using App Router, the `"use client";` directive must be added.
```

It's called a directive :) We'll need to adjust that in all the doc

**Pull request merged**